compiler/ballerina-lang/src/main/java/io/ballerina/projects/JBallerinaBalaWriter.java

```diff
@@ -138,32 +138,28 @@ protected void addCompilerPlugin(ZipOutputStream balaOutputStream) throws IOExce
             List<String> compilerPluginLibPaths = new ArrayList<>();
             List<String> compilerPluginDependencies = this.compilerPluginToml.get().getCompilerPluginDependencies();
 
-            if (compilerPluginDependencies.get(0) == null) {
+            if (compilerPluginDependencies.isEmpty()) {
```

Comment:
@Shadow-Devil The fix is working fine now we get the below error message. 
<img width="748" alt="image" src="https://github.com/user-attachments/assets/31416e4f-95c8-42f8-8c30-b6abc5214e56">

Shall we add a negative test case for this? 
We have a few negative test cases in `io.ballerina.projects.test.plugins.CompilerPluginNegativeTests`. Can you please add one for this fix?